apps/accounts/views.py

```diff
@@ -132,3 +132,16 @@ class ProfileView(JWTProtectedView, generics.RetrieveAPIView):
 
     def retrieve(self, request, *args, **kw):
         return Response(self.get_serializer(request.user).data)
+
+
+class AccountSourceView(generics.RetrieveAPIView):
+    serializer_class = AccountSourceSerializer
+
+    @waffle_switch('fxa-auth')
+    def retrieve(self, request, *args, **kwargs):
+        email = request.GET.get('email')
+        try:
+            user = UserProfile.objects.get(email=email)
+        except UserProfile.DoesNotExist:
+            user = UserProfile(fxa_id='sure')
```

Comment:
This is a bit weird. I was thinking that perhaps the logic should move to `UserProfile.source`. That way this can be an empty user and it can say `source` is `fxa` if there is no `pk`. I'll move it.